xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/adapter/AbstractService.java

```diff
@@ -92,16 +103,28 @@ public void setHttpClient(HttpClient httpClient) {
                     }
 
                     Matcher matcher = CHARSET_PATTERN.matcher(responseHeaders.getHeader(CONTENT_TYPE_HEADER));
+
+                    String charset = null;
+
                     if (matcher.find()) {
-                        return baos.toString(matcher.group(1));
+                        charset = matcher.group(1);
                     }
 
-                    return baos.toString(StandardCharsets.UTF_8.name());
+                    if (charset == null) {
+                        charset = StandardCharsets.UTF_8.name();
```

Comment:
please assign StandardCharsets.UTF_8.name() directly to the charset variable at line #107, after you can remove this unnecessary null check

Reply:
Done, thanks!